### ISSUE TITLE
Connection reset when connecting to Oracle DB over SSH

### ISSUE THREAD

**Issue by fgblomqvist** [serge-rider/dbeaver#1202]
I am trying to set up a connection to an Oracle 11g database via an SSH tunnel, but when I try to connect it just instantly gives me a connection reset error. Manually SSH'ing and connecting to the database works fine.

I'm using the ojdbc6 driver from the page that DBeaver told me to download it from.

What more information can I provide?

**Comment by serge-rider** [comment]
Make sure you set connection properties correctly. Usually database host name should be set to "localhost" when you work thru SSH tunnel.

**Comment by fgblomqvist** [comment]
The database I am trying to connect to is hosted by my university, so the details are a bit blurry.
I normally connect by SSH'ing to one of their linux computers and then connect via "sqlplus <username>@<host>". I am not entirely sure what the database name is though.

**Comment by serge-rider** [comment]
Well, there are two host names in case of SSH: target tunnel host (on SSH settings page) and database host name (on connection settings page). Target tunnel host must be accessible from your local machine.

When you establish SSH tunnel your network context is changed. Now you are on the tunnel target machine. So you have to specify database host name as it is visible from this machine.

To check your host name: connect to your target machine by some SSH software (as you usually do).
run `telenet <host> 1521` where <host> is database host name. If you see "Connection refused" error then you need to use some other Oracle host name (internal network IP address?).

Note: you can't use Oracle TNS with SSH. Only direct host+port.

**Comment by fgblomqvist** [comment]
The telnet command just said that my hostname didn't exist, so I then looked up what the sqlplus command accepts. After digging around the script files that the school created for setting up things (sqlplus and whatnot doesn't work by default, we have to run a command first), I figured out that the hostname we had was actually a SID. I then managed to find the tnsnames.ora file, which contained the actual host address of the database (another machine on the school network, seemingly only accessible from within the network so still have to use the SSH tunnel).
After entering everything in the right place in DBeaver it now works! So happy, thanks a lot for the info!
